packages/test-utils/src/reporter.ts

```diff
@@ -309,32 +322,44 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
     return { approximateDuration, resultCounts };
   }
 
-  getTestId(source?: TestIdContext) {
+  private _getTestId(source?: TestIdContext) {
     if (!source) {
-      return this.baseId;
+      return this._baseId;
     }
 
-    return `${this.baseId}-${[...source.scope, source.title].join("-")}-${source.attempt}`;
+    return `${this._baseId}-${[...source.scope, source.title].join("-")}-${source.attempt}`;
   }
 
-  parseConfig(config: ReplayReporterConfig<TRecordingMetadata> = {}, metadataKey?: string) {
-    this.apiKey = config.apiKey || process.env.REPLAY_API_KEY || process.env.RECORD_REPLAY_API_KEY;
-    this.upload = "upload" in config ? !!config.upload : !!process.env.REPLAY_UPLOAD;
-    if (this.upload && !this.apiKey) {
+  private _parseConfig(
+    config: ReplayReporterConfig<TRecordingMetadata> = {},
+    metadataKey?: string
+  ) {
+    this._apiKey = config.apiKey || process.env.REPLAY_API_KEY || process.env.RECORD_REPLAY_API_KEY;
+    this._upload = "upload" in config ? !!config.upload : !!process.env.REPLAY_UPLOAD;
+    if (this._upload && !this._apiKey) {
       throw new Error(
-        `\`@replayio/${this.runner.name}/reporter\` requires an API key to upload recordings. Either pass a value to the apiKey plugin configuration or set the REPLAY_API_KEY environment variable`
+        `\`@replayio/${this._runner.name}/reporter\` requires an API key to upload recordings. Either pass a value to the apiKey plugin configuration or set the REPLAY_API_KEY environment variable`
       );
     }
+    this._minimizeUploads = config.upload
+      ? typeof config.upload === "object" && typeof config.upload.minimizeUploads === "boolean"
+        ? config.upload.minimizeUploads
+        : false
+      : false;
+    this._uploadStatusThreshold = config.upload
+      ? (typeof config.upload === "object" && config.upload.statusThreshold) || "all"
+      : "none";
```

Comment:
I wonder if we could write this in a way that's a bit less _dense_? I think it's easy to misread the semantics for upload status threshold in particular ("all" vs "none")

I wonder if this would be a little clearer/easier to read? TBH I'm not even sure.

```ts
if (config.upload) {
  if (isAdvancedUploadOption(config.upload)) {
    this._minimizeUploads = config.upload.minimizeUploads ?? false;
    this._uploadStatusThreshold = config.upload.statusThreshold ?? "all";
  } else {
    this._uploadStatusThreshold = "all";
  }
}
```